File: TwoSum.swift
//1. Two Sum - Leetcode Easy

class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        for i in 0...nums.count-1 {
            for j in 0...nums.count-1 {
                if target - nums[j] == nums[i] && i != j {
                    return [i,j]
                }
            }
        }
        return nums
    }
}

File: TwoSum.ts
// 1. Two Sum - LeetCode Easy

function twoSum(nums: number[], target: number): number[] {
    // Edge cases
    // empty nums array?
    // target = 0 ?
    // all +ve numbers ? 
    // if no sum added equal to target, return empty array?
    
    // Solution:
    // nested for loop to add each of the numbers
    // Ex: 2 --> try adding 7, 11, and 15 to it
    for (let i = 0; i < nums.length; i++) {
        for (let j = 0; j < nums.length; j++) {
            if ( i != j && target - nums[j] == nums[i] ) {
                return [i,j];
            }
        }
    }

    return [];
};
